src/commands/githubActions/bumpUpstream/index.ts

```diff
@@ -99,110 +95,80 @@ async function gaBumpUpstreamHandler({
  *
  * @param {string} dir - Directory for the Compose file.
  * @param {Compose} compose - Original Docker Compose configuration.
- * @param {UpstreamRepoMap} upstreamRepoVersions - Mapping of dependencies to their new versions.
- * @return {ComposeVersionsToUpdate | null} - Details of updated versions or null.
+ * @param {UpstreamSettings[]} upstreamSettings - New versions for the Compose services.
  */
-function updateComposeVersions(dir: string, compose: Compose, upstreamRepoVersions: UpstreamRepoMap): ComposeVersionsToUpdate | null {
-  const newCompose = JSON.parse(JSON.stringify(compose)); // Deep copy
-  const versionsToUpdate: ComposeVersionsToUpdate = {};
+function updateComposeVersions(dir: string, compose: Compose, upstreamSettings: UpstreamSettings[]): void {
+  const newCompose: Compose = JSON.parse(JSON.stringify(compose)); // Deep copy
 
-  for (const [serviceName, service] of Object.entries(compose.services)) {
+  for (const upstreamItem of upstreamSettings) {
 
-    if (typeof service.build !== "object" || !service.build.args)
-      continue;
+    for (const [, service] of Object.entries(newCompose.services))
 
-    for (const [argName, currentVersion] of Object.entries(service.build.args)) {
-      const upstreamVersionInfo = upstreamRepoVersions[argName];
+      if (typeof service.build !== "string" && service.build?.args && upstreamItem.arg in service.build.args)
 
-      if (!upstreamVersionInfo || currentVersion === upstreamVersionInfo.newVersion)
-        continue;
-
-      newCompose.services[serviceName].build.args[argName] = upstreamVersionInfo.newVersion;
-
-      versionsToUpdate[upstreamVersionInfo.repoSlug] = {
-        newVersion: upstreamVersionInfo.newVersion,
-        currentVersion,
-      };
-    }
-  }
-
-  if (isEmpty(versionsToUpdate)) {
-    return null;
-  } else {
-    writeCompose(newCompose, { dir });
+        service.build.args[upstreamItem.arg] = upstreamItem.githubVersion;
   }
 
-  return versionsToUpdate;
+  writeCompose(newCompose, { dir });
 }
 
 /**
- * Updates the manifest with a new version and upstream version tag based on the
- * provided `composeVersionsToUpdate`. It also writes the updated manifest to disk.
+ * Updates the manifest with new version tags based on the provided `composeVersionsToUpdate`
+ * and optionally fetches a new version for the manifest. The updated manifest is returned.
+ * @param {Object} options - The options for updating the manifest.
+ * @param {Manifest} options.manifest - The manifest object to update.
+ * @param {UpstreamSettings[]} options.upstreamSettings - The new versions for the manifest.
+ * @param {string} options.dir - Directory path where the manifest will be saved.
+ * @param {string} options.ethProvider - Ethereum provider URL.
+ * @returns {Promise<Manifest>} The updated manifest object.
  */
-async function updateManifest({
+async function updateManifestVersions({
```

Comment:
On first sight this function does the same as `updateComposeVersions`, bumping upstreamVersion, but it does not. It also bumps the pkg itselve version.

These are 2 different actions and was thinking if we should have separated functions for each action, with the corresponding proper naming: `updateManifestUpstreamVersions` and `updateManifestPkgVersion`. Realize that `updateManifestUpstreamVersions` would be sync as `updateComposeUpstreamVersions` and `updateManifestPkgVersion` would be async

Reply:
I agree

Reply:
Done